internal/promotion/runner/builtin/git_cloner.go

```diff
@@ -103,10 +103,24 @@ func (g *gitCloner) run(
 			SSHPrivateKey: creds.SSHPrivateKey,
 		}
 	}
+	var repoUser git.User
+
+	if cfg.User != nil {
```

Comment:
Don't forget to do something similar to this for the `git-commit` step in order to account for https://github.com/akuity/kargo/issues/3813#issuecomment-2828788694.